src/db/sql/events-actions/types.ts

```diff
@@ -1,90 +1,222 @@
+// Namespaces are used here as an alias to the simple number and number[] types in order to model
+// the relationships between the different types of data in the system.
+
+import { AccountUpdate } from 'o1js';
+
+/**
+ * A Field is the smallest unit of raw data in an Event
+ */
+namespace Field {
+  /**
+   * The id of the Field in the archive node postres DB
+   */
+  export type Id = number;
+
+  /**
+   * The raw value of the Field
+   */
+  export type Value = string;
+}
+
 /**
- * Type representing a database row with detailed information related to the archive node.
- * This includes fields such as block-related hashes, account information, action states, and more.
+ * An EventFieldArray is an Array of {@link Field.Id} which represents a multi-Field datum within an event
+ * For instance, a public key is 2 fields, so it will generate an EventFieldArray with 2 Field.Ids
+ */
+namespace EventFieldArray {
+  /**
+   * The id of the EventFieldArray in the archive node postres DB
+   */
+  export type Id = number;
+
+  /**
+   * An array of {@link Field.Id}
+   */
+  export type FieldIds = Field.Id[];
+}
+
+/**
+ * An Event is a collection of {@link EventFieldArray.Id} which represents a single event in the system
+ * Because an event may contain multiple data, it is represented as an array of arrays
+ */
+namespace Event {
+  /**
+   * The id of the Event in the archive node postres DB
+   */
+  export type Id = number;
+
+  /**
+   * An array of {@link EventFieldArray.Id}
+   */
+  export type EventFieldArrayIds = EventFieldArray.Id[];
+}
+
+namespace AccountUpdate {
+  /**
+   * The id of the AccountUpdate in the archive node postres DB
+   */
+  export type Id = number;
+
+  /**
+   * An array of {@link Event.Id}
+   */
+  export type EventIds = Event.Id[];
+}
+
+/**
+ * Represents a complete row from the Archive Node database.
+ * This structure gathers information related to blocks, transactions, events, and associated zkApp updates.
  */
 export type ArchiveNodeDatabaseRow = {
-  // Unique block identifier.
+  /**
```

Comment:
This comment styling makes intellisense render the comments on hover.

![image](https://github.com/user-attachments/assets/7a62f027-bf21-40ee-9d14-940d2772b08f)